pennylane/templates/tensornetworks/mera.py

```diff
@@ -0,0 +1,258 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the MERA template.
+"""
+# pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+import warnings
+import pennylane as qml
+import pennylane.numpy as np
+from pennylane.operation import Operation, AnyWires
+
+
+def compute_indices(wires, n_block_wires):
+    """Generate a list containing the wires for each block.
+
+    Args:
+        wires (Iterable): wires that the template acts on
+        n_block_wires (int): number of wires per block
+
+    Returns:
+        layers (array): array of wire labels for each block
+    """
+
+    n_wires = len(wires)
+
+    if n_block_wires % 2 != 0:
+        raise ValueError(f"n_block_wires must be an even integer; got {n_block_wires}")
+
+    if n_block_wires < 2:
+        raise ValueError(
+            f"number of wires in each block must be larger than or equal to 2; got n_block_wires = {n_block_wires}"
+        )
+
+    if n_block_wires > n_wires:
+        raise ValueError(
+            f"n_block_wires must be smaller than or equal to the number of wires; "
+            f"got n_block_wires = {n_block_wires} and number of wires = {n_wires}"
+        )
+
+    if not np.log2(n_wires / n_block_wires).is_integer():
+        warnings.warn(
+            f"The number of wires should be n_block_wires times 2^n; got n_wires/n_block_wires = {n_wires/n_block_wires}"
+        )
+
+    n_layers = np.floor(np.log2(n_wires / n_block_wires)).astype(int) * 2 + 1
+    wires_list = []
+    wires_list.append(list(wires[0:n_block_wires]))
+    highest_index = n_block_wires
+    for i in range(n_layers - 1):
+        n_elements_pre = 2 ** ((i + 1) // 2)
+        if i % 2 == 0:
+            new_list = []
+            list_len = len(wires_list)
+            for j in range(list_len - n_elements_pre, list_len):
+                new_wires = [
+                    wires[i] for i in range(highest_index, highest_index + n_block_wires // 2)
```

Comment:
maybe good to replace `i` with another index, e.g. `k` to avoid confusion with the outer index in the for loop